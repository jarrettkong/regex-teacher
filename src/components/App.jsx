import React, { Component } from 'react';
import RegexForm from './RegexForm.jsx';
import './App.scss';

class App extends Component {
  render() {
    return (
      <header className="App">
        <h1>/ Regex /</h1>
        <RegexForm />
      </header>
    );
  }
}

export default App;
