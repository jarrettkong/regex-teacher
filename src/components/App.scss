$accent-color: #00C296;
$background: #394950;

.App {
  background-color: $background;
  height: 100vh;
}

.App-header {
  height: 50vh;
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.App-name {
  font-size: 100px;
  margin: 0 0 10px 0;
}

.App-tagline {
  margin: 0 0 25px 0;
}

.App-btn-container {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 25px;
}

.App-btn {
  width: 200px;
  height: 40px;
  border: 0;
  color: white;
  background-color: $accent-color;
  font-size: 17px;
  border-radius: 2px;
  font-weight: 600;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  text-decoration: none;
  box-shadow: 0 0 1px lightgrey;

  &:hover {
    background-color: adjust-color($color: $accent-color, $lightness: 3%)
  }

  &:focus {
    outline: transparent;
  }
}