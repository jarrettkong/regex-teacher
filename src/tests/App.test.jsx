import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App.jsx';
import { shallow } from 'enzyme';

describe('App', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    );
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
})
