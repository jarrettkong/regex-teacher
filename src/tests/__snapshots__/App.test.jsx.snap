// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<section
  className="App"
  id="App"
>
  <BrowserRouter>
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        path="/learn"
        render={[Function]}
      />
      <Route
        path="/sandbox"
        render={[Function]}
      />
    </Switch>
  </BrowserRouter>
</section>
`;
